ast/desugar/Desugar.cc

```diff
@@ -317,8 +317,15 @@ ExpressionPtr buildMethod(DesugarContext dctx, core::LocOffsets loc, core::LocOf
     auto enclosingBlockArg = blockArg2Name(dctx, *blkArg);
 
     DesugarContext dctx2(dctx1.ctx, dctx1.uniqueCounter, enclosingBlockArg, declLoc, name);
-    ExpressionPtr desugaredBody = desugarBody(dctx2, loc, body, std::move(destructures));
-    desugaredBody = validateRBIBody(dctx2, move(desugaredBody));
+
+    bool removeBody = dctx.ctx.state.autogenPrintingSubclassesOrAutoloaderOnly;
+    ExpressionPtr desugaredBody;
+    if (!removeBody) {
```

Comment:
Somewhat surprised to see `!removeBody` instead of simply `removeBody` (with swapped branches) given that there are both an `if` and and `else` branch (in my head it reads a bit like a double negation).

But that's not blocking.

Reply:
Fixed